v3/lints/cabf_ev/lint_invalid_orgid_reg_scheme.go

```diff
@@ -0,0 +1,68 @@
+/*
+ * ZLint Copyright 2024 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+// Contributed by <asantoni64@gmail.com>
+
+package cabf_ev
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+
+	"fmt"
+)
+
+func init() {
+	lint.RegisterCertificateLint(&lint.CertificateLint{
+		LintMetadata: lint.LintMetadata{
+			Name:          "e_ev_invalid_orgid_reg_scheme",
+			Description:   "The Registration Schemes allowed in organizationIdentifier are those listed in Appendix H",
+			Citation:      "CABF EV Guidelines §9.2.8",
+			Source:        lint.CABFEVGuidelines,
+			EffectiveDate: util.CABV170Date,
+		},
+		Lint: NewInvalidOrgIDRegistrationScheme,
+	})
+}
+
+type invalidOrgIDRegistrationScheme struct{}
+
+func NewInvalidOrgIDRegistrationScheme() lint.LintInterface {
+	return &invalidOrgIDRegistrationScheme{}
+}
+
+func (l *invalidOrgIDRegistrationScheme) CheckApplies(c *x509.Certificate) bool {
+	return util.IsEV(c.PolicyIdentifiers) && util.IsSubscriberCert(c) && c.Subject.OrganizationIDs != nil
+}
+
+func (l *invalidOrgIDRegistrationScheme) Execute(c *x509.Certificate) *lint.LintResult {
+
+	// Let's assume there is just one OrganizationID; if not so, it's not this lint's business to raise an alarm
+	orgId := c.Subject.OrganizationIDs[0]
```

Comment:
Would you mind length checking this index as well? I know it's unlikely, but it's the wild west out.

Reply:
I am not sure what you mean, here. The `CheckApplies` function already checks that `c.Subject.OrganizationIDs != nil`, and based on zcrypto/x509/pkix/pkix.go it seems to me that this implies that OrganizationIDs contains at least one element, so I can always take its element [0] ... no?

Reply:
Unfortunately, no. A nil-check protects against a null dereference (except not really because of...Go)**, but it does not protect against indexing into an empty slice.

```go
func main() {
	org := []byte{}
	if org != nil {
		// Explosion
		fmt.Println(org[0])
	}
}
```

** When derefencing a nil pointer to a slice, Go will automagically dereference a zero value instead.

```go
func main() {
	var org []byte = nil
	// true 
	fmt.Println(org == nil)
	// []
	fmt.Println(org)
}
```

Neat, but also confusing.

Reply:
Thank you! I know it's `nitty`, but it always comforts me to know that local code protects itself and is not reliant upon safety hidden within another codebase.